logic/extensions/atrium-logic-kotlin_1_3/src/commonMain/kotlin/ch/tutteli/atrium/logic/kotlin_1_3/impl/DefaultResultAssertions.kt

```diff
@@ -34,7 +34,20 @@ class DefaultResultAssertions : ResultAssertions {
         container: AssertionContainer<out Result<*>>,
         expectedType: KClass<TExpected>
     ):  SubjectChangerBuilder.ExecutionStep<Throwable?, TExpected> =
-        container.manualFeature(EXCEPTION) { exceptionOrNull() }.transform().let { previousExpect ->
+        container.manualFeature(EXCEPTION) {
+
+
+            if(exceptionOrNull() == null && container.maybeSubject.map { exceptionOrNull() } != null){
+                //unwrap somehow (?) todo
```

Comment:
Checkout https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/ to get to know Result better.
Also, `container.maybeSubject.map { exceptionOrNull() }` will never be null but always `Option<Throwable?>` you need to use `getOrElse { null }` in addition to turn it into a `Throwable?`